airbyte-cdk/bulk/core/extract/src/main/kotlin/io/airbyte/cdk/read/FeedReader.kt

```diff
@@ -71,8 +72,15 @@ class FeedReader(
     }
 
     private suspend fun createPartitions(partitionsCreatorID: Long): List<PartitionReader> {
-        val partitionsCreator: PartitionsCreator =
-            root.partitionsCreatorFactory.make(root.stateManager, feed)
+        val partitionsCreator: PartitionsCreator = run {
+            for (factory in root.partitionsCreatorFactories) {
+                log.info { "Attempting bootstrap using ${factory::class}." }
+                return@run factory.make(root.stateManager, feed) ?: continue
+            }
+            throw SystemErrorException(
+                "Unable to bootstrap for feed $feed with ${root.partitionsCreatorFactories}"
+            )
+        }
```

Comment:
Instead of one factory, we inject a list of factories. Micronaut picks up all available implementations of the interface.
We use the first one that doesn't return `null`.